## 1. Знакомство с данными

In [1231]:
import pandas as pd
import numpy as np

In [1232]:
# Чтение данных
url = "autokz2019.csv"
data = pd.read_csv(url, sep=";")
print(data.tail(1).T)

                                                    39965
Год                                                  2019
Месяц                                            Сентябрь
Компания                                         ТК КАМАЗ
Бренд                                               KAMAZ
Модель                                             58815Z
Модификация                                  KAMAZ 58815Z
Год выпуска                                          2019
Страна-производитель                 Республика Казахстан
Вид топлива                                        Дизель
Объём двиг, л,                                        NaN
Коробка передач                                        MT
Тип привода                                           NaN
Сегмент                               Грузовой автомобиль
Регион                                             Актобе
Наименование дилерского центра        ЗапКазКАМАЗ -Актобе
Тип клиента                                      Юр. Лицо
Форма расчета 

In [1233]:
data.info()

<class 'pandas.DataFrame'>
RangeIndex: 39966 entries, 0 to 39965
Data columns (total 25 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   Год                             39966 non-null  int64  
 1   Месяц                           39966 non-null  str    
 2   Компания                        39966 non-null  str    
 3   Бренд                           39966 non-null  str    
 4   Модель                          39966 non-null  str    
 5   Модификация                     36375 non-null  str    
 6   Год выпуска                     39465 non-null  str    
 7   Страна-производитель            39966 non-null  str    
 8   Вид топлива                     36826 non-null  str    
 9   Объём двиг, л,                  35708 non-null  str    
 10  Коробка передач                 36711 non-null  str    
 11  Тип привода                     35677 non-null  str    
 12  Сегмент                         33205 non-n

Всего в датасете до момента очистки - 39966 записей и 25 признаков. По 10 признакам видим пропуски значений, по "Форме расчета" пропущено > 50% данных. В основном характерен строковый тип данных, только у Года - int, а у Количества - float64, что является большой проблемой для обработки числовых данных.

In [1234]:
# Проверим уникальные значения
print(data.nunique())

Год                                  1
Месяц                                9
Компания                            40
Бренд                               46
Модель                             273
Модификация                       1383
Год выпуска                          9
Страна-производитель                19
Вид топлива                         10
Объём двиг, л,                     114
Коробка передач                     83
Тип привода                         26
Сегмент                            119
Регион                              26
Наименование дилерского центра     205
Тип клиента                          9
Форма расчета                       21
Количество                          56
Цена, USD                         3510
Продажа, USD                      4183
Область                             18
Сегментация 2013                     5
Класс 2013                          21
Сегментация Eng                     16
Локализация производства             2
dtype: int64


Рассмотрим подробнее уникальные значения по нашим признакам, чтобы выявить значимость и корректность представленных данных

In [1235]:
data['Компания'].unique()

<StringArray>
[                  'Mercur Auto',             'Автоцентр-Бавария',
                    'БИПЭК АВТО',                         'Вираж',
         'Eurasia Motor Premium',         'Daewoo Bus Kazakhstan',
                'Caspian Motors',                'Каспиан Моторс',
                         'СемАЗ',                 'Astana Motors',
                'Astana Motors ', 'Hyundai Com Trans Kazakhstan ',
      'Nissan Manufacturing RUS',                     'СВС-ТРАНС',
                    'Allur Auto',         'Almaty Motors Premium',
                  'TERRA MOTORS',                      'ТК КАМАЗ',
       'Toyota Motor Kazakhstan',              'Лифан Моторс Рус',
    'MAN Truck & Bus Kazakhstan',                    'Автомир ГК',
                   'Autokapital',                   'Автокапитал',
                       'ММС Рус',                       'MMC RUS',
                  'Mercur Autos',              'УзАвто-Казахстан',
        'Равон Моторс Казахстан',        'Ravon 

Обнаружены дубликаты компаний:
- Mercur Auto и Mercur Autos - это наш целевой автоцентр для исследования
- Caspian Motors и Каспиан Моторс
- Autokapital и Автокапитал
- ММС Рус и MMC RUS
- Равон Моторс Казахстан и Ravon Motors Kazakstan
- Хино Моторс Казахстан и Hino Motors 
- 'Astana Motors ' и 'Astana Motors' (с пробелом/без)
- также Автомир ГК и Автомир-Центр - структуры одного ритейлера
- 'Eurasia Motor Premium'  и  'ТОО "Eurasia Motor Zhaik"' — региональные дилерские предприятия одной сети
- а 'Astana Motors', 'Almaty Motors Premium', 'Hyundai Com Trans Kazakhstan' — входят в холдинг «Астана Моторс»

Явные дубликаты необходимо корректно переименовать для сбора статистики и анализа показателей компаний 

In [1236]:
data['Бренд'].unique()

<StringArray>
[                'Audi',                  'BMW',            'Chevrolet',
               'Daewoo',                 'Ford',                'Foton',
                  'GAZ',              'Hyundai',  'Hyundai Truck & Bus',
             'Infiniti',                'Isuzu',                'Iveco',
                  'Jac',               'Jaguar',                'KAMAZ',
                  'Kia',                 'Lada',           'Land Rover',
                'Lexus',                'Lifan',                  'MAN',
                'Mazda',        'Mercedes-Benz', 'Mercedes-Benz Trucks',
                 'Mini',           'Mitsubishi',               'Nissan',
              'Peugeot',              'Porsche',                'Ravon',
              'Renault',               'Scania',                'Skoda',
               'Subaru',               'Toyota',                  'UAZ',
           'Volkswagen',                'Volvo',                 'Урал',
             'Cadillac',             

Бренды представлены уникальные, но:
- 'Hyundai' и 'Hyundai Truck & Bus' — одна и та же корейская марка, которая специализируется на легковых авто (первая) и на грузовых/автобусах (вторая)
- 'Mercedes-Benz' и 'Mercedes-Benz Trucks' — аналогично, один бренд с разделением на легковые и грузовые авто
- 'KAMAZ' и 'Nefaz' — НЕФАЗ является дочерним предприятием КАМАЗа, и в статистике их объединяют в одну группу
- 'Daewoo' и 'Ravon' — бренд Ravon был создан для замены устаревшей марки Daewoo на рынках СНГ (для моделей Matiz, Nexia, Gentra узбекской сборки), по сути это одни и те же машины

Выглядит логичным решением объединить их

In [1237]:
data['Вид топлива'].unique()

<StringArray>
[       'Бензин',        'Дизель',             nan,       'дизель ',
       'бензин ',        'гибрид',             '2',           '1,6',
             '0',       'Электро', 'Электричество']
Length: 11, dtype: str

Обнаружены дубликаты в столбце "Вид топлива" - одни и те же значения зафиксированы прописными буквами и с использованием заглавной. Необходимо объединить:
- Бензин и бензин
- Дизель и дизель
- Электро и Электричество
- значения 0; 1,6; 2 - некорректны и относятся к категории "неизвестно"

In [1238]:
data['Тип привода'].unique()

<StringArray>
[     'передний',           'FWD',       'quattro',        'Задний',
        'Полный',           'AWD',      'Передний',           '4WD',
         '4х2.2',             nan,           'RWD',          '2 WD',
           '2WD',          '4 WD',           '4X4',           '4х4',
           '4x4',           '4x2',           '4х2',           '4X2',
        'Астана', 'Передний (FF)',             '0',            'FF',
          '#Н/Д',        'полный',       '4Motion']
Length: 27, dtype: str

Схожие проблемы с данными в колонке "Тип привода", необходимо корректно переименовать и объединить данные, выделив 4 группы:
1) Передний привод: передний, Передний, Передний (FF), FWD, FF
2) Задний привод: Задний, RWD, 4х2, 4x2, 4X2, 4х2.2
3) Полный привод: Полный, полный, 4WD,4 WD, AWD, 4х4, 4x4, 4X4, quattro, 4Motion
4) Монопривод (нельзя утверждать к какому приводу относится ось - к переднему или заднему): 2WD, 2 WD

Исключить мусор, попавший по ошибке: Астана, 0, #Н/Д, и nan

In [1239]:
data['Объём двиг, л,'].unique()

<StringArray>
[     '1,4',        '2',        '3',      '4,4',      '6,6',      '1,5',
    '1,596',      '2,4',      '1,7',     '26,7',
 ...
      '4,9',     '2.0h',      '2,9', '400 Л.С.',  '4,98 L,',     '4,98',
   '88 KWH',   '8,4 L,',     '6,7L',      '6,5']
Length: 115, dtype: str

Объем двигателя содержит слишком разнообразные значения - есть с литрами (в разных видах), есть без литров, есть в KWH. Явно нужно будет тщательно обработать значения 

In [1240]:
data['Регион'].unique()

<StringArray>
[          'Алматы',       'Нур-Султан', 'Усть-Каменогорск',
          'Уральск',         'Костанай',         'Павлодар',
            'Семей',          'Шымкент',            'Актау',
           'Атырау',           'Актобе',        'Караганда',
    'Петропавловск',         'Кокшетау',           'Рудный',
        'Кызылорда',            'Тараз',        'Туркестан',
      'Талдыкорган',          'ЭКСПОРТ',         'Кульсары',
         'Каскелен',        'Экибастуз',           'Риддер',
         'Сарыагаш',        'Зыряновск']
Length: 26, dtype: str

На момент 2019г. такие наименования как Нур-Султан и Зыряновск корректны (хотя Зыряновск официально переименовали 3 января, но у нас нет наименования Алтай в таблице, поэтому оставим Зыряновск -> при расширении данных на другие периоды потребуется переименовать). Колонка "ЭКСПОРТ" содержит данные об экспорте в разные регионы - рассмотрим ниже, нет ли перекоса в данных

In [1241]:
# Посчитаем данные по регионам
stats_reg = data.groupby('Регион').size().reset_index(name='Количество')
total = stats_reg['Количество'].sum()
stats_reg['Доля_%'] = (stats_reg['Количество'] / total) * 100
stats = stats_reg.sort_values(by='Количество', ascending=False).reset_index(drop=True)
stats

,Регион,Количество,Доля_%
0,Алматы,11179,27.97
1,Нур-Султан,8244,20.63
2,Шымкент,2880,7.21
3,Костанай,2447,6.12
4,Атырау,2162,5.41
5,Караганда,2076,5.19
6,ЭКСПОРТ,1810,4.53
7,Актау,1495,3.74
8,Уральск,1493,3.74
9,Актобе,1351,3.38


Доля экспорта составляет всего 4,5%, что незначительно в общих объемах, но сильно перевешивает доли небольших регионов. В целом, данные распределены логично - самая большая доля у наиболее населенного региона Алматы, на втором месте с отставанием в 0,7% - Нур-Султан - второй по численности город в стране, далее - выделяется группа "середнячков" - относительно крупные города с диапазоном в 3-7%, остальные - после Усть-Каменогорска не прревышают долю в 2%

In [1242]:
data['Область'].unique()

<StringArray>
[                      'г.Алматы',                   'г.Нур-Султан',
 'Восточно-Казахстанская область',  'Западно-Казахстанская область',
           'Костанайская область',           'Павлодарская область',
     'Южно-Казахстанская область',          'Мангистауская область',
             'Атырауская область',            'Актюбинская область',
         'Карагандинская область',   'Северо-Казахстанская область',
            'Акмолинская область',         'Кызылординская область',
             'Жамбылская область',          'Туркестанская область',
            'Алматинская область',                'Экспорт область']
Length: 18, dtype: str

Интересная ситуация с областями:
- Южно-Казахстанская область и Туркестанская область - одна и та же область (в 2018 ЮКО переименовали в Туркестанскую)
- отсутствует Шымкент как город республиканского значения, наравне с Алматы и Нур-султаном - значит, он по ошибке попал в Южно-Казахстанскую область

In [1243]:
data['Форма расчета'].unique()

<StringArray>
[                             'безналичный',
                                 'наличный',
                                 'Наличная',
                                        nan,
                              'Безналичная',
                             'Без.Наличный',
                                   'кредит',
                                      'б/н',
                             'Безналичная ',
                               'Без оплаты',
                     'Собственные средства',
 'Акция Собственные средства (Trade in 5%)',
                                   'лизинг',
                            'перечислением',
                             'Перечисление',
                                 'trade-in',
                             'безналичный ',
                                      'нал',
                             'обмен+кредит',
                                'обмен+нал',
                                 'наличные',
                         'Отсрочка платеж

Форма расчета может быть полезна для определения каналов продаж и маржинальности. Данные можно сгруппировать в следующие блоки:
- Наличные (Cash): 'наличный', 'Наличная', 'наличные', 'нал', 'Собственные средства'
- Кредит (Credit): 'кредит' (сюда же можно временно относить сложные сделки)
- Trade-In / Обмен: 'trade-in', 'обмен+кредит', 'обмен+нал', 'Акция Собственные средства (Trade in 5%)'
- Безналичный расчет: 'безналичный', 'Безналичная', 'Без.Наличный', 'б/н', 'перечислением', 'Перечисление', 'лизинг', 'Отсрочка платежа'
- Мусор / Ошибки: nan, 'Без оплаты', '0'

In [1244]:
data['Количество'].unique()

array([  1.,   2.,   4.,   3.,   7.,   5.,   8.,  10.,   6.,   9.,  11.,
        19.,  12.,  70.,  29.,  18.,  37.,  17.,  15.,  24.,  14.,  13.,
        25.,  63.,  23.,  30.,  21.,  60.,  nan,  22.,  16.,  35.,  28.,
        32.,  54., 115.,  46.,  33.,  47.,  20.,  66.,  27.,  50.,  36.,
        26.,  31.,  34.,  40.,  43.,  51.,  42.,  38.,  41., 100.,  -1.,
        79.,  62.])

Обнаружили отрицательное значение, что либо ошибка, либо таким образом зафиксирован возврат

In [1245]:
data['Страна-производитель'].unique()

<StringArray>
[            'Германия',                  'США',              'Австрия',
 'Республика Казахстан', 'Российская Федерация',                'Корея',
               'Япония',              'Таиланд',                'Китай',
                   'UK',           'Узбекистан',              'Венгрия',
               'Турция',              'Испания',           'Нидерланды',
               'Польша',               'Швеция',           'Белоруссия',
              'Бельгия']
Length: 19, dtype: str

Рассмотрим, какие данные хранятся в схожих по смыслу столбцах - Сегмент, Сегментация 2013, Сегмментация Eng, Класс 2013, чтобы оставить из них наиболее полезные

In [1246]:
data['Сегментация 2013'].unique()

<StringArray>
[    'Легковые автомобили',            'Внедорожники',
 'Коммерческие автомобили',                'Минивэны',
                  'Пикапы']
Length: 5, dtype: str

In [1247]:
data['Сегментация Eng'].unique()

<StringArray>
[ 'C',  'D',  'E',  'F', 'K2', 'K3', 'G2', 'K1',  'B',  'O', 'L1',  'N',  'M',
 'L2', 'K4',  'A']
Length: 16, dtype: str

In [1248]:
data['Класс 2013'].unique()

<StringArray>
[                  'C класс',                   'D класс',
                   'E класс',                   'F класс',
       'Среднеразмерные SUV',        'Полноразмерные SUV',
     'Спортивные автомобили',            'Компактные SUV',
                   'B класс',         'Субкомпактные SUV',
          'Большие автобусы', 'Крупнотоннажные грузовики',
             'Микроавтобусы',   'Малотоннажные грузовики',
 'Среднетоннажные грузовики',    'Полноразмерный Минивэн',
          'Средние автобусы',                  'Pick-ups',
      'Развозные автомобили',                   'A класс',
                'Компактвэн']
Length: 21, dtype: str

In [1249]:
data['Сегмент'].unique()

<StringArray>
[                  'Легковые автомобили Premium',
                          'Внедорожники Premium',
                                             nan,
                           'Легковые автомобили',
                                      'B класс ',
                          'Среднеразмерные SUV ',
                                 'Внедорожники ',
                                  'Внедорожники',
                           'Полноразмерные SUV ',
                            'Субкомпактные SUV ',
 ...
                                        'PICKUP',
                                         'sedan',
                                       'автобус',
 'Изотермический фургон 5900*2200*2200 Стандарт',
  'Изотермический фургон 5900*2200*2200 Премиум',
                                           'Bus',
                                      'Автобусы',
                                        '5doors',
                                        '5 door',
                               

Самым "грязным" оказался столбец 'Сегмент'. Наиболее лаконично данные сгруппированы в столбце 'Сегментация 2013' - оставим его для анализа в совокупности с 'Класс 2013'.

In [1250]:
data['Наименование дилерского центра'].unique()

<StringArray>
[      'Mercur Auto Алматы',       'Mercur Auto Астана',
 'Автоцентр-Бавария Астана', 'Автоцентр-Бавария Алматы',
 'Автоцентр Бавария Восток',     'ЖАЙЫК HYUNDAI MOTORS',
    'Hyundai Auto Kostanai',      'БИПЭК АВТО Костанай',
        'БИПЭК АВТО Астана',      'БИПЭК АВТО Павлодар',
 ...
          'КВЦ-АВТО Астана',            'КазАгроФинанс',
          'КВЦ-АВТО Алматы',       'КВЦ-АВТО Караганда',
           'КВЦ-Авто Тараз',     'Меридиан АВТО Астана',
  'Меридиан АВТО Караганда',    'Меридиан АВТО Шымкент',
              'СемАЗ Тараз',          'СемАЗ Кызылорда']
Length: 205, dtype: str

In [1251]:
data['Коробка передач'].unique()

<StringArray>
[          'S-tronic',           'S-Tronic',          'Tiptronic',
               'АКПП',                'АКП',         'Steptronic',
                 'AT',               '6 АТ',               '5 МТ',
                '6АТ',                 'MT',                '8AT',
                '5МТ',      'Powershift S6',                '6AT',
                  nan,               'Мех.',               '6 AT',
               '6 MT',                '6MT',               '5 AT',
                '5AT',               'МКПП',              '7 DCT',
               '7DCT',               '8 AT',                 'МТ',
                'CVT',                'DCT',               '4 АТ',
               '6 МТ',                '6МТ',                '4АТ',
                '4AT',                '8АТ',               '8 АТ',
                '5MT',                'AMT',     'CVT (вариатор)',
     'CVT (вариATор)',      'CVT(вAриATор)',          '7G-TRONIC',
          '9G-TRONIC',               '12AT',    

In [1252]:
data['Модель'].unique()

<StringArray>
[                'A3',                 'A4',                 'A6',
                 'A7',                 'A8',                 'Q5',
                 'Q7',            '3 серия',            '5 серия',
            '6 серия',
 ...
           'HFF6850G', 'Transporter Kombi ',          'GLE Coupe',
      'HFF6124G03EV3',        'HFF6127GZ-4',              'Setra',
          'ZK6108HGH',             'Arkana',                'V90',
             '58815Z']
Length: 273, dtype: str

Наблюдаем немало мусорных данных. Удалим некоторые столбцы, которые не несут большого смысла или содержат слишком большое количество пропусков: 
- удалим Наименование дилерского центра - дублирует по смыслу Компания в сочетании с регионом
- Локализация производства - содержит только 2 значения: локальное и импорт, но у нас все ячейки по регионам и областям заполнены, так что отсутствует необходимость по данному признаку заполнять более важные столбцы
- Тип клиента - присутствуют стандартные ФЛ, ЮЛ (корпорат) и пропуски
- Модификация - некий сплав Бренда и Модели, но содержит много шума
- Сегмент, Сегментация Eng, Сегментация 2013 и Класс 2013 по своему смыслу схожи - оставим Сегментацию 2013 и Класс 2013

In [1253]:
# Удаление лишних столбцов
columns_to_drop = [
    'Наименование дилерского центра', 'Форма расчета', 'Сегментация Eng',
    "Сегмент", 'Локализация производства', 'Тип клиента', 'Модификация'
]
data = data.drop(columns=columns_to_drop, errors="ignore")

# Удалим полностью пустые строки
empty_rows = data.shape[0]
data.dropna(how='all', inplace=True)

# Удалим явные дубликаты
data.drop_duplicates(inplace=True)
final_rows = data.shape[0]
print(f"Удалено: {empty_rows - final_rows}")

Удалено: 23471


In [1254]:
# Переименуем оставшиеся признаки
rename_dict = {
    'Год': 'year',
    'Месяц': 'month',
    'Компания': 'dealer_group',
    'Бренд': 'brand',
    'Модель': 'model',
    'Год выпуска': 'release_year',
    'Страна-производитель': 'country',
    'Вид топлива': 'fuel_type',
    'Тип кузова': 'body_type',
    'Объём двиг, л,': 'engine_volume',
    'Коробка передач': 'transmission',
    'Тип привода': 'drive_type',
    'Регион': 'region',
    'Область': 'area',
    'Класс 2013': 'class',
    'Сегментация 2013': 'segment',
    'Количество': 'quantity',
    'Цена, USD': 'price',
    'Продажа, USD': 'income'
}

data.rename(columns=rename_dict, inplace=True)

In [1255]:
data.info()

<class 'pandas.DataFrame'>
Index: 16495 entries, 0 to 39951
Data columns (total 18 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   year           16495 non-null  int64  
 1   month          16495 non-null  str    
 2   dealer_group   16495 non-null  str    
 3   brand          16495 non-null  str    
 4   model          16495 non-null  str    
 5   release_year   16296 non-null  str    
 6   country        16495 non-null  str    
 7   fuel_type      15399 non-null  str    
 8   engine_volume  15065 non-null  str    
 9   transmission   15447 non-null  str    
 10  drive_type     15127 non-null  str    
 11  region         16495 non-null  str    
 12  quantity       16489 non-null  float64
 13  price          16495 non-null  str    
 14  income         16495 non-null  str    
 15  area           16495 non-null  str    
 16  segment        16495 non-null  str    
 17  class          16495 non-null  str    
dtypes: float64(1), int64(1

## Глубокая очистка и преобразование данных
На предыдущем этапе было выявлено много мусора, некорретных типов данных, ошибок - обработаем эти находки

In [1256]:
# Создадим копию датасета
data_clean = data.copy()

In [1257]:
# Преобразуем сразу дробные числа
pd.set_option('display.float_format', lambda x: '%.2f' % x)

In [1258]:
# Сформируем корректную дату - склеим год и месяц
# Создадим словарь для перевода названий месяцев в числа
data_clean['month'] = data_clean['month'].astype(str).str.strip().str.lower()
months_ru_to_num = {
    'январь': '01', 'февраль': '02', 'март': '03', 'апрель': '04',
    'май': '05', 'июнь': '06', 'июль': '07', 'август': '08',
    'сентябрь': '09', 'октябрь': '10', 'ноябрь': '11', 'декабрь': '12'
}

# Переводим текстовый месяц в строковое число
data_clean['month_cleaned'] = data_clean['month'].astype(str).str.strip().str.lower()
data_clean['month_num'] = data_clean['month_cleaned'].map(months_ru_to_num)

# Конкатенируем: Год + Месяц + Число '01' (добавили 1-е, т.к. оно есть во всех месяцах)
data_clean['date_str'] = (
    data_clean['year'].astype(str) + '-' + 
    data_clean['month_num'] + '-01'
)

# Превращаем строку в тип данных datetime
data_clean['date'] = pd.to_datetime(data_clean['date_str'], errors='coerce')

# Переносим дату на последний день месяца
data_clean['date'] = data_clean['date'] + pd.offsets.MonthEnd(0)

# Удаляем временные столбцы
data_clean = data_clean.drop(columns=['month', 'year', 'month_cleaned', 'month_num', 'date_str'])

print(data_clean[['date']].head())

        date
0 2019-05-31
1 2019-08-31
2 2019-04-30
3 2019-07-31
5 2019-02-28


In [1259]:
# Обработка названий компаний
data_clean['dealer_group'] = data_clean['dealer_group'].astype(str).str.strip()

dealer_mapping = {
    'Mercur Autos': 'Mercur Auto',
    'Каспиан Моторс': 'Caspian Motors',
    'Автокапитал': 'Autokapital',
    'ММС Рус': 'MMC RUS',
    'Равон Моторс Казахстан': 'Ravon Motors Kazakstan',
    'Хино Моторс Казахстан': 'Hino Motors',
    'Astana Motors ': 'Astana Motors',
    'Автомир-Центр': 'Автомир ГК',
    'ТОО "Eurasia Motor Zhaik"': 'Eurasia Motor',
    'Eurasia Motor Premium': 'Eurasia Motor'
}
data_clean['dealer_group'] = data_clean['dealer_group'].replace(dealer_mapping)

In [1260]:
# Обработка брендов
data_clean['brand'] = data_clean['brand'].astype(str).str.strip()

brand_mapping = {
    'Hyundai Truck & Bus': 'Hyundai',
    'Mercedes-Benz Trucks': 'Mercedes-Benz',
    'Nefaz': 'KAMAZ',
    'Daewoo': 'Ravon' 
}
data_clean['brand'] = data_clean['brand'].replace(brand_mapping)

In [1261]:
# Обработка стран
data_clean['country'] = data_clean['country'].astype(str).str.strip()

country_mapping = {
    'Германия': 'DEU',
        'США': 'USA',
        'Австрия': 'AUT',
        'Республика Казахстан': 'KAZ',
        'Российская Федерация': 'RUS',
        'Корея': 'KOR',
        'Япония': 'JPN',
        'Таиланд': 'THA',
        'Китай': 'CHN',
        'UK': 'GBR',
        'Узбекистан': 'UZB',
        'Венгрия': 'HUN',
        'Турция': 'TUR',
        'Испания': 'ESP',
        'Нидерланды': 'NLD',
        'Польша': 'POL',
        'Швеция': 'SWE',
        'Белоруссия': 'BLR',
        'Бельгия': 'BEL'
}

data_clean['country'] = data_clean['country'].map(country_mapping).fillna(data_clean['country'])

In [1262]:
# Обработка Вида топлива
data_clean['fuel_type'] = data_clean['fuel_type'].astype(str).str.strip().str.lower()

fuel_mapping = {
    'бензин': 'F',
    'дизель': 'D',
    'электро': 'E', 'электричество': 'E',
    'гибрид': 'HYB'
}
data_clean['fuel_type'] = data_clean['fuel_type'].map(fuel_mapping).fillna('UNKNOWN')

In [1263]:
# Обработка типов привода
data_clean['drive_type'] = data_clean['drive_type'].astype(str).str.strip()

drive_mapping = {
    # Передний
    'передний': 'Передний', 'FWD': 'Передний', 'FF': 'Передний', 'Передний (FF)': 'Передний',
    # Задний
    'Задний': 'Задний', 'RWD': 'Задний', '4х2': 'Задний', '4x2': 'Задний', '4X2': 'Задний', '4х2.2': 'Задний',
    # Полный
    'полный': 'Полный', 'Полный': 'Полный', '4WD': 'Полный', '4 WD': 'Полный', 'AWD': 'Полный', 
    '4х4': 'Полный', '4x4': 'Полный', '4X4': 'Полный', 'quattro': 'Полный', '4Motion': 'Полный',
    # Монопривод
    '2 WD': '2WD', '2WD': '2WD'
}
data_clean['drive_type'] = data_clean['drive_type'].replace(drive_mapping).fillna('Не указано')

In [1264]:
# Функция очистки текста от пробелов, запятых и знаков валют
def to_clean_numeric(series):
    cleaned = series.astype(str).str.replace(r'[\s\s\u200b\u200f₸\$]', '', regex=True)
    cleaned = cleaned.str.replace(',', '.')
    return pd.to_numeric(cleaned, errors='coerce')

# Приводим к числам (если в цене/итоговой стоимости есть копейки — будет float, иначе int)
data_clean['price'] = to_clean_numeric(data_clean['price']).fillna(0.0)
data_clean['income'] = to_clean_numeric(data_clean['income']).fillna(0.0)

# Количество — пропуски заполняем нулями
data_clean['quantity'] = data_clean['quantity'].fillna(0).astype(int)

In [1265]:
# Обработка областей
data_clean['area'] = data_clean['area'].astype(str).str.strip()
data_clean['region'] = data_clean['region'].astype(str).str.strip()

# Исправления значений
# Заменили ЮКО на Туркестанскую
data_clean['area'] = data_clean['area'].replace({'Южно-Казахстанская область': 'Туркестанская область'})
# Добавили г. Шымкент - город рег.значения
data_clean.loc[data_clean['region'] == 'Шымкент', 'area'] = 'г.Шымкент'

print(data_clean['area'].unique())
print(data_clean['region'].unique())

<StringArray>
[                      'г.Алматы',                   'г.Нур-Султан',
 'Восточно-Казахстанская область',  'Западно-Казахстанская область',
           'Костанайская область',           'Павлодарская область',
                      'г.Шымкент',          'Мангистауская область',
             'Атырауская область',            'Актюбинская область',
         'Карагандинская область',   'Северо-Казахстанская область',
            'Акмолинская область',         'Кызылординская область',
             'Жамбылская область',          'Туркестанская область',
            'Алматинская область',                'Экспорт область']
Length: 18, dtype: str
<StringArray>
[          'Алматы',       'Нур-Султан', 'Усть-Каменогорск',
          'Уральск',         'Костанай',         'Павлодар',
            'Семей',          'Шымкент',            'Актау',
           'Атырау',           'Актобе',        'Караганда',
    'Петропавловск',         'Кокшетау',           'Рудный',
        'Кызылорда',   

Теперь все значения учтены, наименования корректные, города и области записаны с большой буквы

In [1266]:
# Обработка отрицательных значений количества
# data_clean['quantity'] = data_clean['quantity'].abs() - если бы это была опечатка
# Фиксируем возврат для выручки
data_clean.loc[data_clean['quantity'] < 0, 'income'] = -data_clean['income'].abs()

In [1267]:
# Обработка объема двигателя
# Базовая очистка (запятые, пробелы, литера L)
data_clean['engine_volume'] = (
    data_clean['engine_volume']
    .astype(str)
    .str.replace(',', '.', regex=False)     # Заменяем запятые на точки
    .str.replace(r'[lLлЛ]', '', regex=True) # Удаляем литеру L (английскую и русскую)
    .str.strip()
)

# Переводим в формат float 
data_clean['engine_volume'] = pd.to_numeric(data_clean['engine_volume'], errors='coerce')

In [1268]:
# Проверим данные по Ягуару и Ниве
print("Максимальный объем у Jaguar:", data_clean[data_clean['brand'] == 'Jaguar']['engine_volume'].max())
print("Объемы для Chevrolet Niva:", data_clean[data_clean['model'].str.contains('Niva', na=False)]['engine_volume'].unique())

Максимальный объем у Jaguar: 5.0
Объемы для Chevrolet Niva: [ 1.7 26.7 25.7 24.7 23.7 22.7 21.7 20.7 19.7 18.7 17.7 16.7 15.7 14.7
 13.7 12.7 11.7 10.7  9.7  8.7  7.7  6.7  5.7  4.7  3.7  2.7]


В данных не обнаружилось, что у Jaguar указан объём 400, максимальный объем - 5л, что соответствует реальным показателям. А вот с Нивой необходимо разобраться. Исторически у Нивы двигатели объемом не более 1.9 литра

In [1269]:
# Исправим объем двигателя для Chevrolet Niva
niva_mask = (data_clean['brand'] == 'Chevrolet') & (data_clean['model'].str.contains('Niva', case=False, na=False))
data_clean.loc[niva_mask & (data_clean['engine_volume'] > 1.9), 'engine_volume'] = 1.7

In [1270]:
# Удалим строки с объемом двигателя = 0.0
data_clean = data_clean[data_clean['engine_volume'] > 0]

# Округляем объемы до 1 знака после запятой, чтобы избавиться от хвостов
data_clean['engine_volume'] = data_clean['engine_volume'].round(1)

In [1271]:
# Еще раз проверим все значения
data_clean['engine_volume'].unique()

array([ 1.4,  2. ,  3. ,  4.4,  6.6,  1.5,  1.6,  2.4,  1.7,  6.2,  2.5,
        2.7,  2.8,  4.3,  3.9,  7.5, 12.3,  3.5,  5.6,  8.7,  5.2,  5. ,
        1.2,  3.3,  1.8,  4.6,  5.7, 10.5,  4. ,  5.5, 12. , 12.8, 11. ,
        2.2, 13. ,  1.3,  3.8,  3.6,  7.6,  4.9,  2.9,  6.7,  6.5])

In [1272]:
# Ообработка типов трансмиссий
data_clean['transmission'] = data_clean['transmission'].astype(str).str.strip().str.lower()

# Создаем словарь
transmission_mapping = {
    # МЕХАНИКА
    'механика': 'Механика', 'мкпп': 'Механика', 'мт': 'Механика', 'мех.': 'Механика', 'мкп': 'Механика',
    'mt': 'Механика', '5 мт': 'Механика', '5мт': 'Механика', '6 mt': 'Механика', '6mt': 'Механика', 
    '6 мт': 'Механика', '6мт': 'Механика', '5mt': 'Механика', '6m/t': 'Механика', 'мt': 'Механика', 
    '5m': 'Механика', '5м': 'Механика', 'm/t': 'Механика', 'м/t': 'Механика', 'mechanic': 'Механика',
    
    # АВТОМАТ
    'автомат': 'Автомат', 'акпп': 'Автомат', 'акп': 'Автомат', 'at': 'Автомат', '6 ат': 'Автомат', 
    '6ат': 'Автомат', '8at': 'Автомат', '6at': 'Автомат', '6 at': 'Автомат', '5 at': 'Автомат', 
    '5at': 'Автомат', '8 at': 'Автомат', '4 ат': 'Автомат', '4ат': 'Автомат', '4at': 'Автомат', 
    '8ат': 'Автомат', '8 ат': 'Автомат', '12at': 'Автомат', '12ат': 'Автомат', '5ат': 'Автомат', 
    '7ат': 'Автомат', '7at': 'Автомат', '8-ступ акпп': 'Автомат', '8акпп': 'Автомат', 'ат': 'Автомат', 
    'аt': 'Автомат', 'а/t': 'Автомат', ' 7 акп': 'Автомат', '7акп': 'Автомат', '6a': 'Автомат', 
    '6а': 'Автомат', '8a': 'Автомат', '9at': 'Автомат', '8': 'Автомат', 'automated': 'Автомат',
    's-tronic': 'Автомат', 's-tronic': 'Автомат', 'tiptronic': 'Автомат', 'steptronic': 'Автомат', 
    'powershift s6': 'Автомат', '7 dct': 'Автомат', '7dct': 'Автомат', 'dct': 'Автомат', 'amt': 'Автомат', 
    'pdk': 'Автомат', '7 dsg': 'Автомат', '7dsg': 'Автомат', '6 dsg': 'Автомат', '6dsg': 'Автомат', 
    'dsg': 'Автомат', '7g-tronic': 'Автомат', '9g-tronic': 'Автомат', '8 акпп tiptronic s': 'Автомат', 
    '7 акпп (pdk)': 'Автомат', '7 акпп pdk': 'Автомат', '8 акпп (pdk)': 'Автомат', 'редуктор': 'Автомат',
    'cvt': 'Автомат', 'cvt (вариатор)': 'Автомат', 'cvt (вариатoр)': 'Автомат', 'cvt(вaриатoр)': 'Автомат'
}

data_clean['transmission'] = data_clean['transmission'].map(transmission_mapping).fillna('Не указано')

In [1273]:
data_clean['transmission'].unique()

<StringArray>
['Автомат', 'Механика', 'Не указано']
Length: 3, dtype: str

In [1274]:
# Обработка классов
data_clean['class'] = data_clean['class'].astype(str).str.strip()
# Переименовываем Pick-ups
data_clean['class'] = data_clean['class'].replace({'Pick-ups': 'Пикапы'})


In [1275]:
# Преобразуем Дату продажи в формат datetime (я бы перевела в int для удобства)
data_clean['release_year'] = pd.to_datetime(data_clean['release_year'], errors='coerce')

In [1276]:
# Переводим текстовые столбцы в категориальный тип 
categorical_columns = [
    'fuel_type',     # Тип топлива
    'transmission',  # Тип трансмиссии
    'drive_type',    # Тип привода
    'segment',       # Сегмент 2013
    'class'          # Класс 2013
]

for col in categorical_columns:
    if col in data_clean.columns:
        data_clean[col] = data_clean[col].astype('category')


У нас остались некоторые пропуски значений: 
- release_year   15978 non-null
- engine_volume  13620 non-null

Т.к. данных по объему двигателя отсутсвует > 5%, то лучше их заполнить медианой на основе группировки по моделям. Год выпуска имеет меньше пропусков, можно было бы удалить, но тоже заполним медианой по моделям

In [1277]:
# Группируем по моделям и заполняем пропуски медианой внутри этой же модели
data_clean['engine_volume'] = data_clean.groupby('model')['engine_volume'].transform(lambda x: x.fillna(x.median()))

# На случай, если есть модели, где пропущена колонка объема, заполним медианой по бренду
data_clean['engine_volume'] = data_clean.groupby('brand')['engine_volume'].transform(lambda x: x.fillna(x.median()))

# Единичные пропуски заполняем общей медианой
data_clean['engine_volume'] = data_clean['engine_volume'].fillna(data_clean['engine_volume'].median())

In [1278]:
# Заполняем пропуски в годах медианным годом для конкретной модели
data_clean['release_year'] = data_clean.groupby('model')['release_year'].transform(lambda x: x.fillna(x.median()))
data_clean['release_year'] = data_clean['release_year'].fillna(data_clean['release_year'].median())

In [1279]:
print(data_clean.info())

<class 'pandas.DataFrame'>
Index: 13618 entries, 0 to 39950
Data columns (total 17 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   dealer_group   13618 non-null  str           
 1   brand          13618 non-null  str           
 2   model          13618 non-null  str           
 3   release_year   13618 non-null  datetime64[us]
 4   country        13618 non-null  str           
 5   fuel_type      13618 non-null  category      
 6   engine_volume  13618 non-null  float64       
 7   transmission   13618 non-null  category      
 8   drive_type     13618 non-null  category      
 9   region         13618 non-null  str           
 10  quantity       13618 non-null  int64         
 11  price          13618 non-null  float64       
 12  income         13618 non-null  float64       
 13  area           13618 non-null  str           
 14  segment        13618 non-null  category      
 15  class          13618 non-null  cate

In [1280]:
print(data_clean.tail())

              dealer_group   brand      model release_year country fuel_type  \
39849           Allur Auto  Yutong  ZK6108HGH   2019-01-01     KAZ         D   
39880           Allur Auto  Yutong  ZK6108HGH   2019-01-01     KAZ         D   
39948  Scandinavian Motors   Volvo        V90   2018-01-01     SWE         F   
39949  Scandinavian Motors   Volvo        V90   2019-01-01     SWE         F   
39950  Scandinavian Motors   Volvo        V90   2019-01-01     SWE         F   

       engine_volume transmission drive_type    region  quantity    price  \
39849           6.70     Механика        2WD  Костанай         1 91979.01   
39880           6.50     Механика        2WD  Костанай         1 98175.23   
39948           2.00      Автомат     Полный    Алматы         1 59199.85   
39949           2.00      Автомат     Полный    Алматы         1 56646.48   
39950           2.00      Автомат     Полный    Алматы         1 60256.41   

        income                  area                  se

## Исследовательский анализ данных (EDA)

In [1281]:
data_clean.describe()

,release_year,engine_volume,quantity,price,income,date
count,13618,13618.00,13618.00,13618.00,13618.00,13618
mean,2018-09-02 00:54:21.095608,2.32,1.93,28232.62,41967.15,2019-05-25 03:55:10.265824
min,2013-01-01 00:00:00,1.20,-1.00,7100.00,-35588.25,2019-01-31 00:00:00
25%,2018-01-01 00:00:00,1.60,1.00,14200.00,17099.33,2019-03-31 00:00:00
50%,2019-01-01 00:00:00,2.00,1.00,20700.00,27228.91,2019-05-31 00:00:00
75%,2019-01-01 00:00:00,2.70,1.00,32904.24,43510.61,2019-07-31 00:00:00
max,2019-01-01 00:00:00,13.00,115.00,305252.02,3778206.26,2019-09-30 00:00:00
std,NaN,0.96,3.71,22323.47,73799.17,NaN


Данные удалось почистить и преобразовать. Рассмотрим числовые признаки.
- Видим, что даты продаж (date) ожидаемо представлены за 9 месяцев с января по сентябрь 2019 года. 
- Год выпуска (release_year) содержит преимущественно новый автопарк, есть машины 2011 года выпуска, но уже 25% показывает отметку - 2018г.
- engine_volume - включает от 1,2 до 13 л объем двигателя
- количество (quantity) содержит отрицательый минимум из-за возврата, максимум = 115 проданных единиц. Но наиболее типична 1 продажа
- Выручка (income) также содержит отрицательный минимум из-за возврата, максимум = 3778206.26, среднее находится на уровне 75% - на отметке 41000. Значения вероятнее всего фиксировались в долларах
- Стоимость - мин = 7100, мах = 328510, т.к. в линейке машин представлены разные классы - от легковых и недорогих, до большегрузов

Расммотрим категориальные признаки

In [1282]:
# Посмотрим структуру продаж по типам трансмиссии в %
transmission_shares = data_clean['transmission'].value_counts(normalize=True) * 100
transmission_shares

transmission
Автомат      67.65
Механика     30.85
Не указано    1.51
Name: proportion, dtype: float64

Автомат занимает лидирующую позицию на рынке, но 31% механики - тоже неплохой показатель заинтересованности клиентов. 

In [1283]:
# Рассмотрим выручку и количество по брендам
brand_analysis = data_clean.groupby('brand', as_index=False).agg(
    Количество_шт=('quantity', 'sum'),
    Выручка_всего=('income', 'sum'),
    Средняя_цена_авто=('price', 'mean')
)

# Сортируем по выручке для выявления лидеров
brand_analysis = brand_analysis.sort_values(by='Выручка_всего', ascending=False).reset_index(drop=True)
brand_analysis.head(5)

,brand,Количество_шт,Выручка_всего,Средняя_цена_авто
0,Lada,11079,107705078.95,10698.44
1,Toyota,2311,96252302.28,41649.63
2,Hyundai,2247,53537438.99,23339.18
3,Kia,2346,45415000.00,21171.93
4,Nissan,1588,38019008.11,24641.02


Наиболее продающаяся машина - Лада с относительно невысокой средней стоимостью - 10700$. Также в лидерах оказались такие бренды как Toyota, Hyundai, Kia и Nissan - их соотношение количество+стоимость дает за меньшее количество проданных машин большую выручку, т.е. это машины уже более дорогого сегмента, доступные не для всех потребительских сегментов.

In [1284]:
# Рассмотрим географию продаж
region_sales = data_clean.groupby('area').agg(
    Штук_продано=('quantity', 'sum'),
    Выручка_всего=('income', 'sum')
).sort_values(by='Выручка_всего', ascending=False)
region_sales['Доля_выручки_%'] = (region_sales['Выручка_всего'] / data_clean['income'].sum()) * 100
region_sales.head(5)

,Штук_продано,Выручка_всего,Доля_выручки_%
area,,,
г.Алматы,5063,154537116.66,27.04
г.Нур-Султан,3680,95365451.89,16.69
Восточно-Казахстанская область,2245,43687097.06,7.64
Атырауская область,2098,38417194.16,6.72
г.Шымкент,1787,34696442.42,6.07


- Больше всего за период 9 месяцев было продано машин в г.Алматы с выручкой 155 млн. и высокой долей на рынке - уверенные 27%. Это логично, т.к. Алматы - самый крупный густонаселенный город Казахстана.
- На втором месте расположился Нур-Султан, который преодолел 15% рубеж по доле выручки и сумел заработать 95 млн.
- Другие регионы уже расположились на уровне <10% доли выручки

In [1285]:
# Рассмотрим технологические тренды по типам топлива
fuel_share = data_clean['fuel_type'].value_counts(normalize=True) * 100
fuel_share

fuel_type
F         97.48
D          2.04
UNKNOWN    0.43
HYB        0.04
Name: proportion, dtype: float64

Подавляющее большинство машин ездит на бензине, 2% машин - на дизеле, 0.04% - гибриды, а электрокары исчезли из нашей подборки (не поняла, где я их потеряла)

In [1286]:
# Выявление оптовых (>10 авто)
opt_deals = data_clean[data_clean['quantity'] >= 10].sort_values(by='quantity', ascending=False)
print(f"Всего: {len(opt_deals)} крупных оптовых сделок")
print(opt_deals[['date', 'brand', 'model', 'region', 'quantity', 'income']].head(5).to_string(index=False))

Всего: 423 крупных оптовых сделок
      date brand   model region  quantity     income
2019-05-31  Lada     4x4  Тараз       115 1035000.00
2019-07-31 Skoda Octavia  Тараз       100 1870000.00
2019-08-31  Lada     4x4 Атырау        66  594000.00
2019-04-30   Kia     Rio Алматы        63  850500.00
2019-09-30   Kia     Rio Алматы        60  810000.00


Из 13618 оставшихся около 3% (423) - достаточно крупные сделки. Среди наших лидеров, видим, что сделки были осуществлены в разные месяцы, по количеству проданных за раз машин лидирует Lada, реализованная в Таразе. На втором месте - сделка в Таразе со Шкодой Октавиа - 100 машин, которая превзошла продажу Lada по выручке на 835 тыс.$. Также наблюдаем удачные сделки Лады в Атырау - 66 штук и продажу Киа Рио в Алматы - по 63 и 60 штук

In [1287]:
# Анализ возвратов
returns = data_clean[data_clean['quantity'] < 0]
print(f"\nВсего: {len(returns)} шт.")
if len(returns) > 0:
    print(returns['brand'].value_counts().head(3))


Всего: 1 шт.
brand
Skoda    1
Name: count, dtype: int64


На текущий момент зафиксирован 1 возврат Шкоды

## Анализ рынка и позиции «Меркур Авто»

In [1288]:
# Создадим флаг, отделяющий "Меркур Авто" от остального рынка
data_clean['is_mercur'] = data_clean['dealer_group'] == 'Mercur Auto'

In [1290]:
# Считаем общие объемы для расчёта долей рынка
total_market_qty = data_clean['quantity'].sum()
total_mercur_qty = data_clean[data_clean['is_mercur']]['quantity'].sum()
mercur_market_share = (total_mercur_qty / total_market_qty) * 100

print(f"Общая доля Меркур Авто: {mercur_market_share:.2f}%")

Общая доля Меркур Авто: 1.43%


Это весьма небольшой процент, который сигнализирует о потребности выделить сильные и слабые стороны, а также сосредоточиться на развитии потенциала компании

In [1292]:
# Анализ в разрезе брендов
brand_analysis = data_clean.groupby(['brand', 'is_mercur'])['quantity'].sum().unstack(fill_value=0)
brand_analysis.columns = ['Рынок_шт', 'Меркур_шт']
brand_analysis['Доля_Меркура_в_бренде_%'] = (brand_analysis['Меркур_шт'] / (brand_analysis['Рынок_шт'] + brand_analysis['Меркур_шт'])) * 100
brand_analysis['Продажи_Меркур_%'] = (brand_analysis['Меркур_шт'] / total_mercur_qty) * 100

print(brand_analysis[brand_analysis['Меркур_шт'] > 0].sort_values(by='Меркур_шт', ascending=False))

            Рынок_шт  Меркур_шт  Доля_Меркура_в_бренде_%  Продажи_Меркур_%
brand                                                                     
Volkswagen       123        277                    69.25             73.87
Porsche            0         52                   100.00             13.87
Audi               0         46                   100.00             12.27


Меркур сосредоточен на продажах машин брендов Volkswagen (73.87%), Porsche (13.87%) и Audi (12.27%). Всего за период было продано 375 машин. И Меркур является монополистом на рынке по сбыту  Porsche и Audi, также высокий процент продаж в отношении конкурентов по моделям Volkswagen. 

Возможно, стоит расширить свой ассортимент и наладить поставки более популярных и продаваемых на рынке машин. Сейчас очевиден упор на продажи брендовых качественных машин - это формирует определенный статус компании, но может отпугивать среднестаистического покупателя, который ориентирован не на качество, а именно на соотношение цена+качество, где цена является нередко определяющим фактором

In [1293]:
# Анализ в разрезе регионов
region_analysis = data_clean.groupby(['area', 'is_mercur'])['quantity'].sum().unstack(fill_value=0)
region_analysis.columns = ['Рынок_шт', 'Меркур_шт']
region_analysis['Доля_рынка_в_регионе_%'] = (region_analysis['Меркур_шт'] / (region_analysis['Рынок_шт'] + region_analysis['Меркур_шт'])) * 100

print(region_analysis.sort_values(by='Меркур_шт', ascending=False).head(5))

                               Рынок_шт  Меркур_шт  Доля_рынка_в_регионе_%
area                                                                      
г.Алматы                           4828        235                    4.64
г.Нур-Султан                       3629         51                    1.39
Атырауская область                 2057         41                    1.95
Карагандинская область             1688         21                    1.23
Западно-Казахстанская область      1342         14                    1.03


Автоцентры Меркура представлены в разных регионах. Стоит отметить, что в густонаселенных Алматы и Нур-Султане также есть центры продаж. Наиболее высокая доля продаж в Алматы, но относительно всего рынка показатели достаточно скромные - 4,64%. А в других регионах не переваливает за 2%

In [1294]:
# Сравним структуру сегментов
segment_analysis = data_clean.groupby(['segment', 'is_mercur'])['quantity'].sum().unstack(fill_value=0)
segment_analysis.columns = ['Рынок_структура_%', 'Меркур_структура_%']
segment_analysis['Рынок_структура_%'] = (segment_analysis['Рынок_структура_%'] / total_market_qty) * 100
segment_analysis['Меркур_структура_%'] = (segment_analysis['Меркур_структура_%'] / total_mercur_qty) * 100

print(segment_analysis)

                         Рынок_структура_%  Меркур_структура_%
segment                                                       
Внедорожники                         38.70               38.13
Коммерческие автомобили               8.25                2.13
Легковые автомобили                  44.08               57.87
Минивэны                              6.65                1.33
Пикапы                                0.88                0.53


В целом, распределение по сегментам выглядит схожим, что по рынку - что у Меркура. Незначительное отставание Меркура обнаружено по продажам внедорожников и пикапов. По легковым машинам есть некоторое преимущество: 57.87% против 44.08%. Коммерческие автомобили и минивэны у Меркура продаются чуть хуже, чем по рынку. Нельзя сказать, что присутствуют аномалии, также стратегия продаж выглядит приемлемой - внедорожники и легковушки осатются самыми продаваемыми

In [1295]:
# Сравним средний чек
price_comparison = data_clean.groupby('is_mercur')['price'].mean()
print(f"Средний чек Меркур Авто: {price_comparison[True]:,.2f}")
print(f"Средний чек по остальному рынку: {price_comparison[False]:,.2f}")

Средний чек Меркур Авто: 39,342.24
Средний чек по остальному рынку: 27,924.06


Средний чек Меркура оказался выше на 11,5 тысяч чека по остальному рынку. Такой высокий средний чек сигнализирует, что холдинг продает меньше единиц товара, но зарабатывает на высокой маржинальности каждой сделки


| 🟢 Сильные стороны (Точки опоры) | 🔴 Слабые стороны (Зона роста) |
| :--- | :--- |
| **Премиальный статус и монополия**<br>Холдинг ассоциируется с немецким качеством, престижем и надежностью. Владельцы Porsche, Audi и топовых моделей Volkswagen привязаны к официальному сервису дилера. | **Критическая зависимость от одного концерна**<br>Весь бизнес и финансовая стабильность холдинга зависят от решений, квот, логистики и ценообразования материнского концерна Volkswagen Group. Любые перебои с поставками бьют по продажам. |
| **Высокомаржинальный сервис**<br>Обслуживание сложных немецких премиум-машин в долгосрочной перспективе приносит автоцентрам огромную прибыль через продажу оригинальных запчастей и высокую стоимость обслуживания. | **Отсутствие самого массового сегмента**<br>Холдинг практически не представлен в самом объемном, быстрорастущем и прибыльном сегменте масс-маркета Казахстана (автомобили стоимостью до 15-20 тыс. долларов). |
| **Фокус на прибыльных сегментах**<br>В структуре продаж дилера преобладают технологичные, дорогие и маржинальные кроссоверы и внедорожники, спрос на которые стабильно высок. | **Слабая представленность в регионах**<br>Бизнес сильно завязан на мегаполисы (Алматы и Нур-Султан). В отдаленных областях и моногородах позиции слабые, так как там нет массового платежеспособного спроса на немецкие авто. |
| **Высокий средний чек сделки**<br>Холдинг продает значительно меньше автомобилей в штуках, чем конкуренты из масс-маркета, но зарабатывает за счет высокой стоимости каждой отдельной проданной машины. | **Слабый B2B-сегмент (Коммерческий транспорт)**<br>Продажи легкой коммерческой техники, автобусов и тяжелых грузовиков у дилера минимальны на фоне крупных игроков рынка, закрывающих корпоративные тендеры. |


In [1296]:
# Сохраним очищенный датафрейм в CSV-файл
data_clean.to_csv('mercur_auto_cleaned.csv', index=False, encoding='utf-8-sig', sep=',')